grazer/src/main/java/edu/asu/diging/grazer/core/fileupload/service/impl/FileUploadServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import edu.asu.diging.grazer.core.domain.ITransformationFilesMetadata;
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesImpl;
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesMetadataImpl;
+import edu.asu.diging.grazer.core.fileupload.db.impl.FileMetadataDatabaseConnection;
+import edu.asu.diging.grazer.core.fileupload.service.IFileUploadService;
+import edu.asu.diging.grazer.web.fileUpload.FileUploadFormImpl;
+
+@Service
+@PropertySource(value = "classpath:config.properties")
+public class FileUploadServiceImpl implements IFileUploadService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private FileMetadataDatabaseConnection connection;
+
+    @Autowired
+    private Environment env;
+    
+    public synchronized String createID() {
+        return UUID.randomUUID().toString();
+    } 
+    
+    private String createDirectory() {
+        File directory = new File(env.getProperty("transformation.file.dir") + File.separator + createID());
+        while(directory.exists()) {
+            directory = new File(env.getProperty("transformation.file.dir") + File.separator + createID());
+        }
+        directory.mkdir();
+        return directory.toString();
+    }
+    
+    private String uploadFiles(CommonsMultipartFile[] multipartFiles) throws IOException {
+
+        String directory = createDirectory();
+
+        for(CommonsMultipartFile file: multipartFiles) {
+            try {      
+                String serverFileName = directory + File.separator + file.getOriginalFilename();
+                File serverFile = new File(serverFileName);
+
+                BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(serverFile));
+                stream.write(file.getBytes());
+                stream.close();
+            } catch(IOException e) {
+                logger.error("You failed to upload " + file.getOriginalFilename(), e);
+                throw new IOException(e);
+            } 
+        } 
+        return directory;
+    }
+
+    @Override
+    public void save(FileUploadFormImpl transformationMetadataAndFiles, Principal principal) throws IOException { 
+        
+        TransformationFilesImpl transformation = new TransformationFilesImpl();
+        TransformationFilesMetadataImpl metadata = new TransformationFilesMetadataImpl();
+        
+        transformation.setTransformationFileContent(transformationMetadataAndFiles.getTransformationFile().getBytes());
+        transformation.setPatternFileContent(transformationMetadataAndFiles.getPatternFile().getBytes());
+        transformation.setTransformationFileName(transformationMetadataAndFiles.getTransformationFile().getOriginalFilename());
+        transformation.setPatternFileName(transformationMetadataAndFiles.getPatternFile().getOriginalFilename());
```

Comment:
what is the transformation object used for?

Reply:
It stores the file names and file contents and it's mapped to a transformationMetadata object

Reply:
No it is never used, should I remove these statements in that case?

Reply:
I made a change which saves the transformation object to the metadata class.